exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/OperatorWrapper.java

```diff
@@ -303,27 +313,27 @@ public void addSummary(TableBuilder tb, Map<String, Long> majorFragmentBusyTally
    * @param operatorType
    * @return index of spill metric
    */
-  private int getSpillCycleMetricIndex(CoreOperatorType operatorType) {
+  private int getSpillCycleMetricIndex(String operatorType) {
     // TODO: DRILL-6642, replace null values for ProtocolMessageEnum with UNRECOGNIZED NullValue to avoid null checks
     if (operatorType == null) {
```

Comment:
Do you think it would be worth addressing DRILL-6642 in this PR?

Reply:
Looks like some other Jira ticket number was meant there since DRILL-6642 is resolved. 
We can't use the non-null value for operator type (it is string now) until we preserve backward compatibility for reading older profiles. With the changes in this PR, we check whether the operator type string is set in the profile, and if not, we will read the operator id that was used before these changes. Having some default value will enforce us to move additional checks to other places, so I don't think that we should address that ticket until we use both values. Maybe for Drill 2.0 when we will be able to break the compatibility, it may be addressed.